app/controllers/api/v1/journeys_controller.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class JourneysController < ApplicationController
+      skip_before_action :verify_authenticity_token
+
+      def create
+        card = find_card
+        unless card
+          render(json: { error: 'Card not found' }, status: :not_found)
+          return
+        end
+
+        journey = build_journey(card)
+        charge_for_journey(card, journey)
+
+        if journey.save
+          render json: journey.card, status: :created
+        else
+          render json: journey.errors, status: :unprocessable_entity
+        end
+      end
+
+      def update
+        card = find_card
+        unless card
+          render(json: { error: 'Card not found' }, status: :not_found)
+          return
+        end
+        journey = find_ongoing_journey(card)
+        return unless journey
+
+        complete_journey(journey)
+
+        if journey.save
+          RefundCalculatorService.new(journey).process_refund
+          render json: journey.card
+        else
+          render json: journey.errors, status: :unprocessable_entity
+        end
+      end
+
+      private
+
+      def find_card
+        Card.find_by(id: params[:card_number])
+      end
+
+      def build_journey(card)
+        if params[:bus_journey] && params[:bus_journey] == 'true'
```

Comment:
Shouldn't be enough with just be a truthy param?
```suggestion
   params[:bus_journey] == 'true'
```